core/src/main/java/hudson/FilePath.java

```diff
@@ -2259,8 +2259,8 @@ public Integer invoke(File base, VirtualChannel channel) throws IOException {
                         public void visit(File f, String relativePath) throws IOException {
                             if (f.isFile()) {
                                 File target = new File(dest, relativePath);
-                                mkdirsE(target.getParentFile());
-                                Util.copyFile(f, writing(target));
+                                Files.createDirectories(target.getParentFile().toPath());
```

Comment:
The calls to `mkdirsE` and `writing` implement the file access rules part of the agent to master security system, so you shouldn't remove them. See https://wiki.jenkins.io/display/JENKINS/Slave+To+Master+Access+Control#SlaveToMasterAccessControl-Fileaccessrules.